engine/lib/notification.php

```diff
@@ -276,44 +271,36 @@ function _elgg_notifications_smtp_default_message_id_header($hook, $type, $retur
 	
 	$mt = microtime(true);
 	
-	$returnvalue['headers']['Message-ID'] = "{$url_path}.default.{$mt}@{$hostname}";
+	$email->addHeader('Message-ID', "{$url_path}.default.{$mt}@{$hostname}");
 	
-	return $returnvalue;
+	return $email;
 }
 /**
  * Adds default thread SMTP headers to group messages correctly.
  * Note that it won't be sufficient for some email clients. Ie. Gmail is looking at message subject anyway.
  *
- * @param string $hook        Equals to 'email'
- * @param string $type        Equals to 'system'
- * @param array  $returnvalue Array containing fields: 'to', 'from', 'subject', 'body', 'headers', 'params'
- * @param array  $params      The same value as $returnvalue
+ * @param string      $hook   "prepare"
+ * @param string      $type   "system:email"
+ * @param \Elgg\Email $email  Email instance
  * @return array
  * @access private
  */
-function _elgg_notifications_smtp_thread_headers($hook, $type, $returnvalue, $params) {
+function _elgg_notifications_smtp_thread_headers($hook, $type, $email) {
 
-	if (!is_array($returnvalue) || !is_array($returnvalue['params'])) {
```

Comment:
Why not check if `$email` is an actual `\Elgg\Email` like you did in `_elgg_notifications_smtp_default_message_id_header`, just in case somebody else breaks stuff